cmd/rook/ceph/operator.go

```diff
@@ -91,14 +99,176 @@ func startOperator(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		rook.TerminateFatal(err)
 	}
-
 	rookImage := rook.GetOperatorImage(context.Clientset, containerName)
 	serviceAccountName := rook.GetOperatorServiceAccount(context.Clientset)
 	op := operator.New(context, volumeAttachment, rookImage, serviceAccountName)
+	if isSecretPresent(context) {
+		initWebhook(context)
+	}
 	err = op.Run()
 	if err != nil {
 		rook.TerminateFatal(errors.Wrapf(err, "failed to run operator\n"))
 	}
 
 	return nil
 }
+
+func isSecretPresent(context *clusterd.Context) bool {
+	_, err := context.Clientset.CoreV1().Secrets("rook-ceph").Get("rook-admission-controller-secret", metav1.GetOptions{})
+	if err != nil {
+		logger.Infof("rook-admission-controller-secret not found. webhook initialization aborted")
+		return false
+	}
+	return true
+}
+
+func initWebhook(context *clusterd.Context) {
+
+	updateDeploymentWithVolumes(context)
+
+	if isMounted, err := isTLSVolumeMounted(context); isMounted && err == nil {
+		createWebhookService(context)
+		createValidatingWebhookConfig(context)
+		webhook.StartServer(context)
+	}
+
+}
+
+func isTLSVolumeMounted(context *clusterd.Context) (bool, error) {
+	rookDeploymentsClient := context.Clientset.AppsV1().Deployments("rook-ceph")
+
+	result, getErr := rookDeploymentsClient.Get("rook-ceph-operator", metav1.GetOptions{})
```

Comment:
just use `err`.

Reply:
Fixed ! I had different names to identify the type of error. Is it required for it to be err everywhere ?